### PR TITLE
Enforce react-testing-library eslint rules

### PR DESCRIPTION
**Pull request by JanKaczmarkiewicz**
related to https://github.com/gravitational/teleport/pull/15813
https://kentcdodds.com/blog/common-mistakes-with-react-testing-library
https://testing-library.com/docs/ecosystem-eslint-plugin-testing-library/

### CODE REVIEW

**Review by ryanclark — APPROVED** [review]

**Review by ibeckermayer — APPROVED** [review]

**Review by michellescripts — APPROVED** [review]

**Review by hatched — APPROVED** [review]

File: packages/design/src/SlideTabs/SlideTabs.test.tsx [hunk 1]
```diff
@@ -15,30 +15,39 @@
...
       <SlideTabs
         tabs={['aws', 'automatically', 'manually']}
         onChange={() => {}}
       />
     );
-    expect(container.getElementsByTagName('label')).toHaveLength(3);
+
+    expect(screen.getByLabelText('aws')).toBeInTheDocument();
+    expect(screen.getByLabelText('automatically')).toBeInTheDocument();
+    expect(screen.getByLabelText('manually')).toBeInTheDocument();
```
Comment (kimlisa) [inline packages/design/src/SlideTabs/SlideTabs.test.tsx:35]: hm, i think this is a different test from the original test? you're testing if these items were rendered, while the original is testing the correct number of expected items. we should keep both?

**Review by kimlisa — COMMENTED** [review]
thanks for cleaning up our tests :) 

@rudream i would give a cursory glance to get an idea of how we should be testing from now on

File: packages/teleport/src/Discover/LoginTrait/LoginTrait.test.tsx [hunk 2]
```diff
@@ -83,29 +74,24 @@ describe('login trait comp behavior', () => {
...
-    let r: RenderResult;
-    await act(async () => {
-      r = render(Component);
-    });
+    render(<Component />);
 
     // Test existing logins to be rendered.
-    let checkboxes: NodeListOf<HTMLInputElement> =
-      r.container.querySelectorAll('input:checked');
-    expect(checkboxes).toHaveLength(2);
```
[the remark this answers is not in the document]
Comment (JanKaczmarkiewicz) [reply packages/teleport/src/Discover/LoginTrait/LoginTrait.test.tsx:94]: restored in 0e6d717f0b0e58a4a402c6cce758f9afadb0683e

File: packages/teleport/src/Discover/LoginTrait/LoginTrait.test.tsx [hunk 3]
```diff
@@ -83,29 +74,24 @@ describe('login trait comp behavior', () => {
...
+    expect(screen.getByLabelText('apple')).toBeChecked();
 
     // Test existing logins to be rendered with a new login name.
     fireEvent.click(screen.getByText(/add an OS/i));
     const inputEl = screen.getByPlaceholderText('name');
     fireEvent.change(inputEl, { target: { value: 'carrot' } });
     fireEvent.click(screen.getByText('Add'));
 
-    checkboxes = r.container.querySelectorAll('input:checked');
-    expect(checkboxes).toHaveLength(3);
```
[the remark this answers is not in the document]
Comment (JanKaczmarkiewicz) [reply packages/teleport/src/Discover/LoginTrait/LoginTrait.test.tsx:105]: restored in https://github.com/gravitational/webapps/commit/0e6d717f0b0e58a4a402c6cce758f9afadb0683e

**Pull request merged**